src/main/java/org/gridsuite/modification/server/modifications/NetworkModificationApplicator.java

```diff
@@ -51,31 +58,86 @@ public class NetworkModificationApplicator {
 
     @Getter private final FilterService filterService;
 
+    private final ExecutorService applicationExecutor;
+
     @Value("${impacts.collection-threshold:50}")
     @Setter // TODO REMOVE when VoltageInitReportTest will no longer use NetworkModificationApplicator
     private Integer collectionThreshold;
 
     public NetworkModificationApplicator(NetworkStoreService networkStoreService, EquipmentInfosService equipmentInfosService,
-                                         ReportService reportService, FilterService filterService) {
+                                         ReportService reportService, FilterService filterService,
+                                         @Value("${max-large-concurrent-applications}") int maxConcurrentApplications) {
         this.networkStoreService = networkStoreService;
         this.equipmentInfosService = equipmentInfosService;
         this.reportService = reportService;
         this.filterService = filterService;
+        this.applicationExecutor = Executors.newFixedThreadPool(maxConcurrentApplications);
     }
 
+    /* This method is used when creating, inserting, moving or duplicating modifications
+     * Since there is no queue for these operations and they can be memory consuming when the preloading strategy is large
+     * (for example for VOLTAGE_INIT_MODIFICATION),
+     * we limit the number of concurrent applications of these modifications to avoid out of memory issues.
+     * We keep the possibility to apply small or medium modifications immediately in parallel without limits.
+     * And if in the future we also need to limit the memory consumption of medium modifications we can add more code here.
+     * Note : we currently have 3 sizes of modifications :
+     * small : preloadingStrategy = NONE
+     * medium : preloadingStrategy = COLLECTION
+     * large : preloadingStrategy = ALL_COLLECTIONS_NEEDED_FOR_BUS_VIEW
+     */
     public NetworkModificationResult applyModifications(List<ModificationInfos> modificationInfosList, NetworkInfos networkInfos, ReportInfos reportInfos) {
+        PreloadingStrategy preloadingStrategy = modificationInfosList.stream()
+            .map(ModificationInfos::getType)
+            .reduce(ModificationType::maxStrategy)
+            .map(ModificationType::getStrategy)
+            .orElse(PreloadingStrategy.NONE);
```

Comment:
Not important: we already have the same code in other places, put it in a Util ?

Reply:
I didn't want to touch the other places to avoid creating regressions or bugs in this PR